src/gui/sdlmain.cpp

```diff
@@ -2923,28 +2923,48 @@ static void show_warning(char const * const message) {
 #endif // WIN32
 }
 
-static void launcheditor() {
-	std::string path,file;
+static int LaunchEditor()
+{
+	std::string path, file;
 	Cross::CreatePlatformConfigDir(path);
 	Cross::GetPlatformConfigName(file);
 	path += file;
-	FILE* f = fopen(path.c_str(),"r");
+
+	FILE *f = fopen(path.c_str(), "r");
 	if (!f && !control->PrintConfig(path)) {
-		printf("tried creating %s. but failed.\n", path.c_str());
-		exit(1);
+		fprintf(stderr, "Tried creating '%s', but failed.\n", path.c_str());
+		return 1;
```

Comment:
`fopen`'ing file with `read` attribute, but warnings says we're trying to create the file.

Reply:
That's because we create this file few lines earlier. From user perspective this interaction makes sense (but I agree, that it's weird looking at the code).

Reply:
Ahh ok! I just wasn't seeing the whole picture; thank you :-) Resolving.